lib/index.js

```diff
@@ -45,18 +41,25 @@ Embark = {
   },
 
   deployContracts: function(env, contractFiles, destFile, chainFile, withProvider, withChain, cb) {
+    var web3 = this.web3;
+    web3.setProvider(new web3.providers.HttpProvider("http://" + this.blockchainConfig.rpcHost + ":" + this.blockchainConfig.rpcPort))
```

Comment:
that's definitely a potential area to go...however, I definitely think this PR needs updating. It's kind of a hack job in retrospect. It could be much cleaner.